### PR TITLE
Create new Dockerfile for builds

### PR DESCRIPTION
**Pull request by proudh**
Submitting Dockerfile for builds for purposes of review

### CODE REVIEW

**Comment by akutz** [comment]
Hi @proudh,

Nice work! Please be aware that some of this will be invalidated by a pending merge from the PR #474. For example, the go-bindata section will be made irrelevant. Please rebase your work on the branch used by #474 and again on `master` once it is merged. Thank you!

Ultimately the goal is, aside from installing the OS and Go, all you should have to do is call `make deps && make`.

**Comment by cantbewong** [comment]
I think that the some changes are needed to build rom and deb packages using this Dockerfile:
using make rpm-all and make deb-all

1. install these additional packages fakeroot libfakeroot alien

2. modify line 379 of the Makefile from:
setarch $(V_ARCH) rpmbuild -ba --quiet \

to

setarch $(V_ARCH) fakeroot rpmbuild -ba --quiet \

I did not try building packages outside a Docker container, but I suspect that these changes may be advisable even for non-Docker builds - I will defer to akutz as the authority on this though

**Comment by akutz** [comment]
Hi @cantbewong / @proudh,

Please be aware you need to rebase this onto the new `master` as well as the `Makefile` has changed quite a bit. Also, I'm not sure you need to modify the `Makefile`. It works the way it is today, so I'm not going to change it just for the Dockerfile when it builds fine both locally and on Travis. Not unless I hear a compelling argument to do so.

**Comment by akutz** [comment]
Hi @cantbewong / @proudh,

Let me rephrase that. I'm not going to change the `rpmbuild` command in the `Makefile` for the Dockerfile unless I see that it also continues to work locally outside of Docker as well on Travis outside of Docker. If it does, then I have no issue with the change @cantbewong suggested.

**Comment by proudh** [comment]
Since the Makefile has changed significantly since this Dockerfile, I'll update the Dockerfile and any changes to the Makefile as appropriate.
